File: Grind169/Week2/67-Add-Binary/solution.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        s = []
        hold = 0
        i = len(a) - 1
        j = len(b) - 1

        while i >= 0 or j >= 0 or hold:
            if i >= 0:
                hold += int(a[i])
                i -= 1
            if j >= 0:
                hold += int(b[j])
                j -= 1

            s.append(str(hold % 2))

            hold = hold // 2

        return "".join(reversed(s))

File: Grind169/Week2/67-Add-Binary/solution.cpp
#include <string>
#include <math.h>

using namespace std;



class Solution {
public:
    string addBinary(string a, string b) {
        string ans;

        int carry=0;
        int i=a.size()-1;
        int j=b.size()-1;

        while (i>=0 || j>=0 || carry>=0) {
            if (i>=0) {
                carry+=(int)(a[i]);
                i--;
            }
            if (j>=0){
                carry+=(int)(b[j]);
                j--;
            }
            ans+=to_string(carry%2);
            carry/=2;
        }

        reverse(ans.begin(),ans.end());
        return ans;
    }
};